.github/workflows/check_interface_compatibility.yml

```diff
@@ -0,0 +1,76 @@
+name: Check interface compatibility
+
+on:
+  pull_request_target:
+
+permissions:
+  pull-requests: write
+
+jobs:
+  check-interface-compatibility:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Extract new public interfaces
+        run: |
+          make init
+          bin/public_interface.py extract >> ${{ runner.temp }}/interfaces.new.json
+
+      - name: Backup bin/public_interface.py
+        run: |
+          # Keep a copy of bin/public_interface.py to use on the base branch
+          # So we are using the same bin/public_interface.py to process old/new codebase.
+          cp bin/public_interface.py ${{ runner.temp }}/public_interface.py
+
+      - name: Checkout the base branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.base_ref }}
+
+      - name: Extract original public interfaces
+        run: |
+          echo "Original HEAD: $(git rev-parse --short HEAD)"
+          make init
+          # Recover bin/public_interface.py
+          cp ${{ runner.temp }}/public_interface.py bin/public_interface.py
+          bin/public_interface.py extract >> ${{ runner.temp }}/interfaces.original.json
+
+      - name: Archive interface files for debugging
+        uses: actions/upload-artifact@v3
+        with:
+          name: interface-json-files
+          path: ${{ runner.temp }}/interfaces.*.json
+          retention-days: 7  # Only storing it for a week in case of debugging needs.
+    
+      - name: Detect compatibility breaking changes
+        id: detect
+        run: |
+          output=$(bash -c '(bin/public_interface.py check ${{ runner.temp }}/interfaces.original.json ${{ runner.temp }}/interfaces.new.json); exit $?')
+          exit_code=$?
+          echo "output=$output"
+          echo "exit_code=$exit_code"
+          echo "output=$output" >> $GITHUB_OUTPUT
+          echo "exit_code=$exit_code" >> $GITHUB_OUTPUT
+
+      - name: Post GitHub PR comment
```

Comment:
Why can't we just let it fail? We use auto-merge so much this won't work unless it's enforced.

Reply:
discussed offline, since it is not enforced a comment catches reviewers attention than a small cross, and it is more readable

Reply:
Why not enforce?

Reply:
As it could be intentional

Reply:
And this homemade script could generate false positives